lib/SILGen/SILGenExpr.cpp

```diff
@@ -2397,9 +2397,32 @@ RValue RValueEmitter::visitCaptureListExpr(CaptureListExpr *E, SGFContext C) {
   return visit(E->getClosureBody(), C);
 }
 
+static OpaqueValueExpr *opaqueValueExprToSubstituteForAutoClosure(
+    const AbstractClosureExpr *e) {
+  // When we find an autoclosure that just calls an opaque closure,
+  // this is a case where we've created the opaque closure as a
+  // stand-in for the autoclosure itself. Such an opaque closure is
+  // created when we have a property wrapper's 'init(wrappedValue:)'
+  // taking an autoclosure argument.
+  if (auto ace = dyn_cast<AutoClosureExpr>(e)) {
+    if (auto ce = dyn_cast<CallExpr>(ace->getSingleExpressionBody())) {
```

Comment:
This seems really, really brittle. Is there a way we can keep this in sync with the synthesis code?

Reply:
I haven't come up with a good way, no. In general we really need a better way to pull out the initial value expression from the type-checked AST for the initialization of the backing storage, because that code also needed too much tweaking. Maybe there's some better way to grab the expression at the right point in `CSApply.cpp` so we can save it.